csrc/device_lower/analysis/predicate_elimination.cpp

```diff
@@ -185,10 +185,16 @@ class ProducerConsumerPairAnalyzer : public OptOutDispatch {
 
     auto pairwise_map = PairwiseLogicalDomainMap(producer, consumer);
     auto c2p =
-        BestEffortReplay::replayPasC(producer, consumer, -1, pairwise_map)
+        BestEffortReplay::replayPasC(
+            producer, consumer, /*consumer_compute_at_axis=*/-1, pairwise_map)
             .getReplay();
-
-    ProducerConsumerPairAnalyzer analyzer(c2p);
+    for (auto [c, p] : pairwise_map.mapConsumerToProducer()) {
+      // replayPasC skips mapping after the compute at position
+      c2p[c] = p;
+    }
```

Comment:
I still don't 100% understand why this is necessary but some cases failed without this. I had thought that `replayPasC(...).getReplay()` would include `pairwise_map.mapConsumerToProducer()` as a subset.